### ISSUE TITLE
Switch `poc-sdk-otel` to `main`

### ISSUE THREAD

**Issue by pellared** [open-telemetry/opentelemetry-dotnet-instrumentation#226]
## Why

Have a common agreement if the design proposed in `poc-sdk-otel` branch is how we plan to evolve the project so that we can use it as the `main` branch.

_There would be a separate issue to track what we would consider as ALPHA or even BETA release. We are probably missing good user documentation._

## What is included

_TODO_

## What scenarios were checked

_TODO_

## What is missing

_TODO_

**Comment by pellared** [comment]
@RassK Can you confirm that the instrumentation works also for SNK-signed applications and on IIS?

**Comment by nrcventura** [comment]
Yes. We should also call out aspnetcore as a separate thing that we also support.
